radiation/python/radsw/swrad_gt4py.py

```diff
@@ -0,0 +1,1751 @@
+    locdict_gt4py["zrdnd"],
+    locdict_gt4py["zb11"],
+    locdict_gt4py["zb22"],
+    locdict_gt4py["jb"],
+    locdict_gt4py["ib"],
+    locdict_gt4py["ibd"],
+    locdict_gt4py["NGB"],
+    locdict_gt4py["idxsfc"],
+    locdict_gt4py["itind"],
+    locdict_gt4py["fxupc"],
+    locdict_gt4py["fxdnc"],
+    locdict_gt4py["fxup0"],
+    locdict_gt4py["fxdn0"],
+    locdict_gt4py["ftoauc"],
+    locdict_gt4py["ftoau0"],
+    locdict_gt4py["ftoadc"],
+    locdict_gt4py["fsfcuc"],
+    locdict_gt4py["fsfcu0"],
+    locdict_gt4py["fsfcdc"],
+    locdict_gt4py["fsfcd0"],
+    locdict_gt4py["sfbmc"],
+    locdict_gt4py["sfdfc"],
+    locdict_gt4py["sfbm0"],
+    locdict_gt4py["sfdf0"],
+    locdict_gt4py["suvbfc"],
+    locdict_gt4py["suvbf0"],
+    domain=shape_nlp1,
+    origin=default_origin,
+    validate_args=validate,
+)
+
+finalloop(
+    indict_gt4py["idxday"],
+    indict_gt4py["delp"],
+    locdict_gt4py["fxupc"],
+    locdict_gt4py["fxdnc"],
+    locdict_gt4py["fxup0"],
+    locdict_gt4py["fxdn0"],
+    locdict_gt4py["suvbf0"],
+    locdict_gt4py["suvbfc"],
+    locdict_gt4py["sfbmc"],
+    locdict_gt4py["sfdfc"],
+    locdict_gt4py["ftoauc"],
+    locdict_gt4py["ftoadc"],
+    locdict_gt4py["ftoau0"],
+    locdict_gt4py["fsfcuc"],
+    locdict_gt4py["fsfcdc"],
+    locdict_gt4py["fsfcu0"],
+    locdict_gt4py["fsfcd0"],
+    outdict_gt4py["upfxc_t"],
+    outdict_gt4py["dnfxc_t"],
+    outdict_gt4py["upfx0_t"],
+    outdict_gt4py["upfxc_s"],
+    outdict_gt4py["dnfxc_s"],
+    outdict_gt4py["upfx0_s"],
+    outdict_gt4py["dnfx0_s"],
+    outdict_gt4py["htswc"],
+    outdict_gt4py["htsw0"],
+    outdict_gt4py["htswb"],
+    outdict_gt4py["uvbf0"],
+    outdict_gt4py["uvbfc"],
+    outdict_gt4py["nirbm"],
+    outdict_gt4py["nirdf"],
+    outdict_gt4py["visbm"],
+    outdict_gt4py["visdf"],
+    locdict_gt4py["rfdelp"],
+    locdict_gt4py["fnet"],
+    locdict_gt4py["fnetc"],
+    locdict_gt4py["fnetb"],
+    locdict_gt4py["flxuc"],
+    locdict_gt4py["flxdc"],
+    locdict_gt4py["flxu0"],
+    locdict_gt4py["flxd0"],
+    locdict_gt4py["heatfac"],
+)
+
+end = time.time()
+print(f"Elapsed time = {end-start}")
+
+outdict_final = dict()
+valdict_final = dict()
+
+for var in outvars:
+    if var in ["htswc", "cldtausw", "htsw0"]:
+        outdict_final[var] = outdict_gt4py[var][:, :, 1:].view(np.ndarray).squeeze()
+    elif var == "htswb":
+        if lhswb:
+            outdict_final[var] = (
+                outdict_gt4py[var][:, :, 1:, :].view(np.ndarray).squeeze()
+            )
+    else:
+        outdict_final[var] = outdict_gt4py[var].view(np.ndarray).squeeze()
+
+    if lhswb:
+        valdict_final[var] = serializer.read(
+            var, serializer.savepoint["swrad-out-000000"]
+        )
+    else:
+        if var != "htswb":
+            valdict_final[var] = serializer.read(
+                var, serializer.savepoint["swrad-out-000000"]
+            )
+
+compare_data(outdict_final, valdict_final)
+
+if do_test:
```

Comment:
we might want to name this 'do_subtests' or 'unit_tests' (though they aren't really unit tests, since this is already doing a test.

Reply:
will update